biogeochem/EDCanopyStructureMod.F90

```diff
@@ -1541,307 +1537,223 @@ subroutine leaf_area_profile( currentSite )
-                   elseif(min_chite < maxh(iv).and.min_chite > minh(iv).and.max_chite > maxh(iv))then
-                      frac_canopy(iv) = (maxh(iv) -min_chite ) / (currentCohort%hite*prt_params%crown_depth_frac(ft)  )
-                      ! this is the layer with the top of the canopy in it.
-                   elseif(max_chite > minh(iv).and.max_chite < maxh(iv).and.min_chite < minh(iv))then
-                      frac_canopy(iv) = (max_chite - minh(iv)) / (currentCohort%hite*prt_params%crown_depth_frac(ft))
-                   elseif(max_chite < maxh(iv).and.min_chite > minh(iv))then !the whole cohort is within this layer.
-                      frac_canopy(iv) = 1.0_r8
-                   endif
-
-                   ! no m2 of leaf per m2 of ground in each height class
-                   currentPatch%tlai_profile(1,ft,iv) = currentPatch%tlai_profile(1,ft,iv) + frac_canopy(iv) * &
-                        currentCohort%lai
-                   currentPatch%tsai_profile(1,ft,iv) = currentPatch%tsai_profile(1,ft,iv) + frac_canopy(iv) * &
-                        currentCohort%sai
+          call UpdatePatchLAI(currentPatch)
 
-                   !snow burial
-                   if(currentSite%snow_depth  > maxh(iv))then
-                      fraction_exposed = 0._r8
-                   endif
-                   if(currentSite%snow_depth < minh(iv))then
-                      fraction_exposed = 1._r8
-                   endif
-                   if(currentSite%snow_depth >= minh(iv) .and. currentSite%snow_depth <= maxh(iv)) then !only partly hidden...
-                      fraction_exposed = 1._r8 - max(0._r8,(min(1.0_r8,(currentSite%snow_depth-minh(iv))/dh)))
-                   endif
-
-                   currentPatch%elai_profile(1,ft,iv) = currentPatch%tlai_profile(1,ft,iv) * fraction_exposed
-                   currentPatch%esai_profile(1,ft,iv) = currentPatch%tsai_profile(1,ft,iv) * fraction_exposed
+          ! -----------------------------------------------------------------------------
+          ! Standard canopy layering model.
+          ! Go through all cohorts and add their leaf area
+          ! and canopy area to the accumulators.
+          ! -----------------------------------------------------------------------------
 
-                enddo ! (iv) hite bins
+          currentCohort => currentPatch%shortest
+          do while(associated(currentCohort))
+             ft = currentCohort%pft
+             cl = currentCohort%canopy_layer
 
-                currentCohort => currentCohort%taller
+             ! ----------------------------------------------------------------
+             ! How much of each tree is stem area index? Assuming that there is
+             ! This may indeed be zero if there is a sensecent grass
+             ! ----------------------------------------------------------------
 
-             enddo !currentCohort
+             if( (currentCohort%treelai+currentCohort%treesai) > 0._r8)then
```

Comment:
To be consistent with other places in the code, should we use variable "nearzero" instead of 0._r8?

Reply:
this has been changed to use nearzero